core/templates/services/debouncer.service.ts

```diff
@@ -27,29 +27,25 @@ export class DebouncerService {
   // be invoked. The function only gets executed after it stops being called
   // for `wait` milliseconds.
   debounce(func: () => void, millisecsToWait: number): () => void {
-    let timeout;
-    let context = this;
-    let args = arguments;
-    let timestamp;
-    let result;
+    let timeout: ReturnType<typeof setTimeout> | null;
```

Comment:
Can't we have an exact type for this variable?

Reply:
Actually, I tried many variables but nothing was working here so at last tried this.

Reply:
The approach should not be a trial and error here. Try to read the code or find ways to trigger this function from the UI and use the dev tools to find the type of the variable.

Can you please check the function's return/variable assigned to this filed in this file and based on that you can conclude the type of this variable?

Let me know what you have tried, I can help you with better suggestions (if needed). :)

Reply:
I made some changes in the latest commit. PTAL!